.github/workflows/test-changed-firestore.yml

```diff
@@ -29,16 +33,104 @@ jobs:
           yarn
       - name: build
         run: yarn build:changed firestore
-      - name: Run tests if firestore or its dependencies has changed
-        run: yarn test:changed firestore
+      - name: Archive build
+        if: ${{ !cancelled() }}
+        run: |
+          tar -cf build.tar .
```

Comment:
nit: you can probably combine the `tar` and `gzip` commands into a single line by specifying `z` to tar, namely

`tar -zcf build.tar.gz .`

Reply:
This leads to tar complaining the directory changed while archiving, and it only happens with CI.

Reply:
This could probably be resolved by building the tar.gz file in a different directory, such as `${{ runner.temp }}/build.tar.gz`